### ISSUE TITLE
Difficult to add project manually to liberty dashboard

### ISSUE THREAD

**Issue by turkeylurkey** [OpenLiberty/liberty-tools-vscode#206]
When you need to add a project to the liberty dashboard manually it is hard to find where to click. 
Here is a portion of the Explorer:
![image](https://user-images.githubusercontent.com/24705144/207185767-57a2409e-f5af-4487-9e64-43273a437599.png)
Section 1 is the project name. You would think you'd click here to add it to liberty but no. If it did work here no list of projects would be needed.
Section 2 is all the files in the project. This list is normally quite long and fills the entire window. Adding the "Add to Liberty Dashboard" menu item here is ok but perhaps not ideal. Maybe pom.xml is special and the menu item could be added for that one file. 
Section 3 is the only area it works. Quite a small area and not normally visible.
Section 4 is another area you might expect to right click to get the context menu and see the "Add to Liberty Dashboard" menu item

**Comment by TrevCraw** [comment]
Expected behaviour is ability to manually add project to Liberty dashboard from sections 1 and 4.
If selecting from section 4 (the space in the Liberty dashboard), expected behaviour would be to prompt users on which project they would like to add.
If selecting from section 1 (the project root folder), expected behaviour would be that we add the selected project to the dashboard - no project prompt required.

**Comment by anusreelakshmi934** [comment]
Based on my understanding, it appears that the required functionality doesn't have direct APIs involved in VSCode. Currently, only the explorerResourceIsFolder and explorerResourceIsRoot conditions are available. To address this issue we may need to implement a custom solution. 
For the specific case mentioned in section 4, we have to verify that the selected item is a folder and it contains the necessary build files. If the check returns true, indicating the presence of build files, you can then update the "when" condition  for "add/remove project to Liberty dashboard" action in the package.json file accordingly.

 In summary, we have to implement a custom function to check for build files in the selected folder and dynamically adjust the when condition in the package.json file based on the result of this check.

**Comment by anusreelakshmi934** [comment]
I have written a function `hasPomXmlOrBuildGradle()` which checks wheather a pom.xml or build.gradle file is present inside the folder. if its found it returns true otherwise false. I have also registered this command in extension.ts file as - `vscode.commands.registerCommand('myExtension.hasPomXmlOrBuildGradle', () => devCommands.hasPomXmlOrBuildGradle()),`. 
And in the package.json file, liberty.dev.add.project is called when hasPomXmlOrBuildGradle() returns true.

**Comment by anusreelakshmi934** [comment]
The function hasPomXmlOrBuildGradle() doesnt seem to work properly and we are not able to verify if a folder contains either a pom.xml or build.gradle file eventhough function is then registered in the extension.ts file. May need to further look into whether the implementation of the function is wrong or not.

**Comment by anusreelakshmi934** [comment]
Progress on this issue is saved on this branch: - [https://github.com/anusreelakshmi934/liberty-tools-vscode/tree/feature%23206](https://github.com/anusreelakshmi934/liberty-tools-vscode/tree/feature%23206 ). 
This change has added a three dots in the corner of liberty dashboard that has - "Add project to Liberty Dashboard" and "Remove project from Liberty Dashboard" . 
Attaching ss below :
<img width="630" alt="image" src="https://github.com/OpenLiberty/liberty-tools-vscode/assets/141303561/41efe110-345f-49d9-93f0-7683ae8e9740">

But to implement a custom function to check for build files in the selected folder is not currently in working condition. needs further look in implementing the function
